packages/ontario-frontend-cli/core/validation/validatePackageJson.js

```diff
@@ -10,15 +11,10 @@ const path = require('path');
  * 
  * @returns {Promise<boolean>} - Returns a promise that resolves to true if the package.json file exists, false otherwise.
  */
-async function doesPackageJsonExist(dir = process.cwd()) {
-    try {
-      await fs.access(path.join(dir, 'package.json'));
-      return true;
-    } catch (err) {
-      logger.error(`Error checking for package.json in ${dir}: ${err.message}`);
-      return false;
-    }
-  }
+const doesPackageJsonExist = withErrorHandling(async (dir = process.cwd()) => {
+  await fs.access(path.join(dir, 'package.json'));
+  return true;
+});
```

Comment:
I believe we need to capture some kind of fail condition here with an error message.

I created a new project, renamed the `package.json` file so that it wouldn't be detetced, and tried to run the `ontario-add-package` command.

I got this as the output:
```
my-frontend-project % ../ontario-add-package.js eslint
/Users/kylebinger/Development/ontario-frontend/packages/ontario-frontend-cli/core/errors/errorHandler.js:13
    throw new CustomError(fn.name, args, error.message);
          ^

GeneralError: Error in function : 
    at /Users/kylebinger/Development/ontario-frontend/packages/ontario-frontend-cli/core/errors/errorHandler.js:13:11
    at async Command.<anonymous> (/Users/kylebinger/Development/ontario-frontend/packages/ontario-frontend-cli/bin/ontario-add-package.js:45:14) {
  functionName: ''
}
```

However in below constant `isOntarioFrontendProject` I get the expected `logger.error()` message as the output.

Reply:
Good flag, Kyle. I have updated the error handling and validation for these checks